sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLListener.scala

```diff
@@ -56,7 +56,10 @@ case class SparkListenerSQLExecutionStart(
 }
 
 @DeveloperApi
-case class SparkListenerSQLExecutionEnd(executionId: Long, time: Long)
+case class SparkListenerSQLExecutionEnd(
+    executionId: Long,
+    time: Long,
+    errorMessage: Option[String] = None)
```

Comment:
Doesn't this break binary compatibility? I think it should have overridden constructor.

Reply:
This is not a public class. It generates JSON string where backward compatibility matters, and this PR added tests for it.